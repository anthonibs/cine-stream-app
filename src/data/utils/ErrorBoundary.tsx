import { Component, ErrorInfo, ReactNode } from 'react';

interface ErrorBoundaryProps {
	children: ReactNode;
}

interface ErrorBoundaryState {
	hasError: boolean;
}

class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
	constructor(props: ErrorBoundaryProps) {
		super(props);
		this.state = { hasError: false };
	}

	handleReload = () => window.location.reload();

	handleBack = () => window.location.replace('/');

	componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
		this.setState({ hasError: true });
		console.error('Erro encontrado: ', error, errorInfo);
	}

	render(): ReactNode {
		if (this.state.hasError) {
			return (
				<div>
					<h2>
						Oops! Um erro inesperado aconteceu. Por favor recarregue a página ou volte ao inicio.
					</h2>
					<img src='/not_found.webp' alt='sem resultado' />
					<button onClick={this.handleReload}>
						<h3>Recarregar Página</h3>
					</button>
					<button onClick={this.handleBack}>
						<h3>Voltar ao Inicio</h3>
					</button>
				</div>
			);
		}

		return this.props.children;
	}
}

export default ErrorBoundary;
