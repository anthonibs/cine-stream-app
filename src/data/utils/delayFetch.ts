export default function delayFetch(ms = 1000) {
	return new Promise((resolve) => setTimeout(resolve, ms));
}
