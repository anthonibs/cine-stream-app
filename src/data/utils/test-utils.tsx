import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';

import { RenderOptions, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { AuthProvider } from 'data/contexts/Auth';
import { LanguageProvider } from 'data/contexts/Language';
import { MyFavoritesProvider } from 'data/contexts/MyFavoritesList';

import { ThemeProvider } from 'styled-components';

import GlobalStyles from 'ui/styles/GlobalStyles';
import dark from 'ui/styles/themes/dark';

interface ProvidersProps {
	children: ReactElement;
}

const Providers = ({ children }: ProvidersProps) => {
	return (
		<ThemeProvider theme={dark}>
			<GlobalStyles />
			<AuthProvider>
				<LanguageProvider>
					<MyFavoritesProvider>
						<BrowserRouter>{children}</BrowserRouter>
					</MyFavoritesProvider>
				</LanguageProvider>
			</AuthProvider>
		</ThemeProvider>
	);
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
	render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
