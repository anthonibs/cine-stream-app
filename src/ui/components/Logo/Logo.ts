import styled from 'styled-components';

export const LogoContainer = styled.figure`
	width: 80px;
	height: 75px;
`;

export const LogoImage = styled.img`
	display: block;
	width: 100%;
	pointer-events: none;
	height: 100%;
	object-fit: cover;
	user-select: none;
`;
