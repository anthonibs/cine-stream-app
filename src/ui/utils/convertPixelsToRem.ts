const pixelsDefault = 16;

export default function convertPixelsToREM(pixel: number): string {
	return `${pixel / pixelsDefault}rem`;
}