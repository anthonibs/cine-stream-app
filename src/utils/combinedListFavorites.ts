import { IAlreadyWatched, IMovie, ITvMovie } from 'data/interfaces';

export function combinedListFavorites<T extends ITvMovie | IMovie>(
	list: T[],
	listFavorite: T[],
	watchList: IAlreadyWatched[]
) {
	const combinedList = list?.map((movie: T) => {
		// Verifica se o filme está na lista de favoritos
		const isFavorite = listFavorite.some((favorite) => favorite.id === movie.id);
		const isWatched = watchList.some((watched) => watched.id === movie.id);

		// Cria um novo objeto com as informações do filme e se ele é um favorito ou não
		return {
			...movie,
			isFavorite: isFavorite,
			alreadyWatched: isWatched,
		};
	});
	return combinedList;
}
