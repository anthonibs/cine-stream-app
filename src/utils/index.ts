export * from './convertMinutesToHours';
export * from './combinedListFavorites';
export * from './convertPixelsToRem';
export * from './removeAccentsFromText';
export * from './differenceBetweenDates';
