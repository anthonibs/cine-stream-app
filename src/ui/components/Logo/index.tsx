import { memo } from 'react';

import logo from 'assets/svgs/logo-cine-stream.svg';

import * as S from './Logo';

const Logo = () => {
	return (
		<S.LogoContainer>
			<S.LogoImage src={logo} alt='Logo oficial da CineStream' loading='lazy' decoding='async' />
		</S.LogoContainer>
	);
};

export default memo(Logo);
