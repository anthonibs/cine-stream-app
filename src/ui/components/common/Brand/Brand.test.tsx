import { render, screen } from 'data/utils/test-utils';
import Logo from '.';

describe('Render component <Logo/>', () => {
	it('The alt attribute is present in the img tag', () => {
		render(<Logo />);
		const logo = screen.getByRole('img');

		expect(logo).toBeInTheDocument();

		expect(logo).toHaveAttribute('alt', 'Logo oficial da CineStream');
	});

	it('Lazy loading is present', () => {
		render(<Logo />);
		const logo = screen.getByRole('img');

		expect(logo).toHaveAttribute('loading', 'lazy');
		expect(logo).toHaveAttribute('decoding', 'async');
	});

	it('Styles have not changed', () => {
		render(<Logo />);

		const logo = screen.getByRole('img');
		expect(logo).toMatchSnapshot('Styles');
	});
});
