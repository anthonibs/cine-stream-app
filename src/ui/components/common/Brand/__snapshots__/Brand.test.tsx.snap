// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component <Logo/> Styles have not changed: Styles 1`] = `
<img
  alt="Logo oficial da CineStream"
  class="sc-gTRrQi bgmLBl"
  decoding="async"
  loading="lazy"
  src="logo-cine-stream.svg"
/>
`;
